src/io_handler.c

```diff
@@ -41,13 +49,30 @@ int extractsf_handle_args(int argc, char ** argv, bool * use_fmt, int * nevents,
     return check_root_filename(* input_file);
 }
 
-int hipo2root_handle_args(int argc, char ** argv, char ** input_file, int * run_no) {
+int hipo2root_handle_args(int argc, char ** argv, char ** input_file, int * run_no, bool * use_simul) {
+    // Handle optional arguments.
+    int opt;
+    while ((opt = getopt(argc, argv, "s")) != -1) {
+        switch (opt) {
+            case 's': * use_simul = true;         break;
+            default:  return 1; // Bad usage of optional arguments.
+        }
+    }
+
+    // Handle positional arguments
     if (argc < 2) return 1;
     if (argc > 3) return 2;
```

Comment:
I'm not 100% on this, but shouldn't this condition change due to the new optional argument? Does this run ok?

Reply:
nvm, it does run.